src/main/java/com/launchdarkly/client/EventProcessor.java

```diff
@@ -14,22 +15,34 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Random;
 import java.util.concurrent.*;
 
 class EventProcessor implements Closeable {
-  private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor(new DaemonThreadFactory());
+  ThreadFactory threadFactory = new ThreadFactoryBuilder()
```

Comment:
This thread factory doesn't actually need to be a class member. The only reason it's assigned here is that we're not instantiating the `scheduler` in the ctor. I would suggest moving the initialization there so we don't unnecessarily keep a reference to the thread factory.